### ISSUE TITLE
downloadBlobToBuffer batchTransferOptions.operation function doesn't check for errors after running blobURL.Download

### ISSUE THREAD

**Issue by wbh1** [Azure/azure-storage-blob-go#134]
### Which version of the SDK was used?
v0.0.0-20181022225951-5152f14ace1c

### Which platform are you using? (ex: Windows, Linux, Debian)
Linux

### What problem was encountered?
panic: runtime error: invalid memory address or nil pointer dereference
```
level=info ts=2019-05-31T03:41:16.250869767Z caller=compact.go:908 msg="start sync of metas"
level=info ts=2019-05-31T03:41:16.522751016Z caller=compact.go:914 msg="start of GC"
level=info ts=2019-05-31T03:41:16.524491465Z caller=compact.go:920 msg="start of compaction"
level=info ts=2019-05-31T03:41:16.615530225Z caller=compact.go:218 msg="compaction iterations done"
level=info ts=2019-05-31T03:41:16.615588744Z caller=compact.go:225 msg="start first pass of downsampling"
level=info ts=2019-05-31T03:41:32.912774859Z caller=compact.go:231 msg="start second pass of downsampling"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x9020ee]

goroutine 10395792 [running]:
github.com/Azure/azure-storage-blob-go/azblob.(*DownloadResponse).Body(0x0, 0x0, 0x0, 0x0, 0x0, 0xc00076b408)
	/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.0.0-20181022225951-5152f14ace1c/azblob/zz_response_helpers.go:62 +0x24e
github.com/Azure/azure-storage-blob-go/azblob.downloadBlobToBuffer.func1(0x0, 0x745, 0x435901, 0x1260101)
	/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.0.0-20181022225951-5152f14ace1c/azblob/highlevel.go:208 +0x162
github.com/Azure/azure-storage-blob-go/azblob.doBatchTransfer.func2(0xc0006b0360, 0xc000419fa8)
	/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.0.0-20181022225951-5152f14ace1c/azblob/highlevel.go:329 +0x3b
github.com/Azure/azure-storage-blob-go/azblob.doBatchTransfer.func1(0xc0006b0360, 0xc0006b05a0)
	/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.0.0-20181022225951-5152f14ace1c/azblob/highlevel.go:312 +0x3f
created by github.com/Azure/azure-storage-blob-go/azblob.doBatchTransfer
	/go/pkg/mod/github.com/!azure/azure-storage-blob-go@v0.0.0-20181022225951-5152f14ace1c/azblob/highlevel.go:309 +0x191
```

(This was encountered in the [Thanos project](https://github.com/improbable-eng/thanos) )

### How can we reproduce the problem in the simplest way?
Make blobURL.Download fail in the `downloadBlobToBuffer` 

### Have you found a mitigation/solution?
Yes. I'll submit a PR for this shortly after submitting this issue.

**Comment by zezha-msft** [comment]
Hi @wbh1, thanks a lot for catching this!

**Comment by zezha-msft** [comment]
The fix will be deployed with the next release. :shipit:

**Comment by wbh1** [comment]
Closing, as this is fixed in v0.7.0
